pkg/router/template/router.go

```diff
@@ -398,6 +399,12 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 	config := ServiceAliasConfig{
 		Host: host,
 		Path: route.Spec.Path,
+		Hash: backendKey,
+	}
+
+	// Possibly hash the name for later use in obsuring cookie ID
+	if os.Getenv("COOKIE_PREFIX_MAIN") + os.Getenv("COOKIE_PREFIX_EDGE") + os.Getenv("COOKIE_PREFIX_REENCRYPT") != "" {
```

Comment:
Look at the integration tests in `test/integration/router_test.go` - what you need to do is probably similar to `TestRouterPathSpecificity` - check that the response back from the client has a cookie + its name is "mangled".